src/main/java/com/mongodb/hibernate/jdbc/MongoConnectionProvider.java

```diff
@@ -106,49 +102,32 @@ public <T> T unwrap(Class<T> unwrapType) {
     }
 
     @Override
-    public void configure(Map<String, Object> configValues) {
-        var jdbcUrl = configValues.get(JAKARTA_JDBC_URL);
+    public void configure(Map<String, Object> configProperties) {
+        var jdbcUrl = configProperties.get(JAKARTA_JDBC_URL);
 
-        if (mongoClientCustomizer == null && jdbcUrl == null) {
+        if (mongoDialectConfigurator == null && jdbcUrl == null) {
             throw new HibernateException(format(
                     "Configuration property [%s] is required unless %s is provided",
-                    JAKARTA_JDBC_URL, MongoClientCustomizer.class.getName()));
+                    JAKARTA_JDBC_URL, MongoDialectConfigurator.class.getName()));
         }
 
-        var connectionString = jdbcUrl == null ? null : getConnectionString(jdbcUrl);
-
-        final MongoClientSettings.Builder clientSettingsBuilder;
-        if (mongoClientCustomizer != null) {
-            clientSettingsBuilder = MongoClientSettings.builder();
-            mongoClientCustomizer.customize(clientSettingsBuilder, connectionString);
-        } else {
-            clientSettingsBuilder = MongoClientSettings.builder().applyConnectionString(connectionString);
+        try {
+            MongoDialectSettings.Builder mongoDialectSettingsBuilder = MongoDialectSettings.builder(configProperties);
+            if (mongoDialectConfigurator != null) {
+                mongoDialectConfigurator.configure(mongoDialectSettingsBuilder);
+            }
+            config = mongoDialectSettingsBuilder.build();
+        } catch (RuntimeException e) {
+            throw new HibernateException(
```

Comment:
Do we have to throw `HibernateException` here? If yes, why?

Reply:
@NathanQingyangXu, sorry, forgot to ping you right away for ☝️

Reply:
That is Hibernate's internal exception base class (other exception like `ServiceException` extends it) so Hibernate has elegant exception handling. Usually `RuntimeException` should be wrapped up somewhere. You can see its Javadoc at https://github.com/hibernate/hibernate-orm/blob/main/hibernate-core/src/main/java/org/hibernate/HibernateException.java